### ISSUE TITLE
[Bug]: Could not access folders whose name include "++" via web interface.

### ISSUE THREAD

**Issue by bindungszustandsamplitude** [nextcloud/server#32512]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Hello together,

since the recent major upgrade to 24.0.0 I could not access folders which include "++" in their name via web interface. Accessing them via direct link is not possible as well (blank page). Accessing them via externally shared link is possible, though - only the internal link seems to be broken.

Please confirm / disconfirm this issue.

### Steps to reproduce

1. Open web interface.
2. Create folder with "++" in its name, e.g. "test++"
3. Open the folder via web interface by clicking on it.
4. The folder does not open and one is stuck on the parent directory.

### Expected behavior

The folder should open and the content should be visible.

### Installation method

Manual installation

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "XXX",
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "24.0.0.12",
        "overwrite.cli.url": "https:\/\/dickerbauch.ddns.net",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "simpleSignUpLink.shown": false,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "auth.bruteforce.protection.enabled": true,
        "theme": "",
        "loglevel": 2,
        "mysql.utf8mb4": true,
        "updater.release.channel": "stable",
        "check_data_directory_permissions": true,
        "app.mail.verify-tls-peer": false,
        "app.mail.imap.timeout": 50,
        "app.mail.smtp.timeout": 20,
        "debug": false,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "default_phone_region": "DE"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - calendar: 3.3.0
  - circles: 24.0.0
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contacts: 4.1.0
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - dav: 1.22.0
  - extract: 1.3.5
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_accesscontrol: 1.14.0
  - files_pdfviewer: 2.5.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - forms: 2.5.0
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - mail: 1.12.0
  - maps: 0.1.10
  - nextcloud_announcements: 1.13.0
  - notes: 4.3.1
  - notifications: 2.12.0
  - oauth2: 1.12.0
  - password_policy: 1.14.0
  - photos: 1.6.0
  - polls: 3.6.1
  - previewgenerator: 5.0.0
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - quota_warning: 1.14.0
  - recommendations: 1.3.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - spreed: 14.0.1
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - tasks: 0.14.4
  - text: 3.5.1
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - twofactor_totp: 6.3.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - viewer: 1.8.0
  - weather_status: 1.4.0
  - workflowengine: 2.6.0
Disabled:
  - admin_audit
  - bruteforcesettings: 2.3.0
  - drawio: 1.0.2
  - encryption
  - files_automatedtagging: 1.13.0
  - files_external: 1.9.0
  - files_texteditor: 2.14.0
  - firstrunwizard: 2.7.0
  - groupfolders: 11.1.2
  - music: 1.5.1
  - passwords: 2022.4.20
  - phonetrack: 0.7.0
  - user_ldap
```

### Nextcloud Signing status

```shell
No errors have been found.
```

**Comment by solracsf** [comment]
Should be fixed in v24.0.1. Please upgrade, test and report back.

**Comment by johnnyq** [comment]
Can anyone confirm this was fixed? If so please close when confirmed

**Comment by bindungszustandsamplitude** [comment]
I can confirm this bug was fixed with v24.0.1. Therefore, I close this issue.
